azkaban-common/src/test/java/azkaban/executor/ExecutorHealthCheckerTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+* Copyright 2019 LinkedIn Corp.
+*
+* Licensed under the Apache License, Version 2.0 (the “License”); you may not
+* use this file except in compliance with the License. You may obtain a copy of
+* the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an “AS IS” BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations under
+* the License.
+*/
+package azkaban.executor;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+import azkaban.Constants.ConfigurationKeys;
+import azkaban.alert.Alerter;
+import azkaban.utils.Pair;
+import azkaban.utils.Props;
+import azkaban.utils.TestUtils;
+import com.google.common.collect.ImmutableMap;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Test case for executor health checker.
+ */
+public class ExecutorHealthCheckerTest {
+
+  private static final int EXECUTION_ID_11 = 11;
+  private static final String AZ_ADMIN_ALERT_EMAIL = "az_admin1@foo.com,az_admin2@foo.com";
+  private final Map<Integer, Pair<ExecutionReference, ExecutableFlow>> activeFlows = new HashMap<>();
+  private ExecutorHealthChecker executorHealthChecker;
+  private Props props;
+  private ExecutorLoader loader;
+  private ExecutorApiGateway apiGateway;
+  private Alerter mailAlerter;
+  private AlerterHolder alerterHolder;
+  private ExecutableFlow flow1;
+  private Executor executor1;
+
+  @Before
+  public void setUp() throws Exception {
+    this.props = new Props();
+    this.props.put(ConfigurationKeys.AZKABAN_EXECUTOR_MAX_FAILURE_COUNT, 1);
```

Comment:
Would it be better to set to 2, to test the counting?

Reply:
Sure.